api/files/permissions.py

```diff
@@ -36,3 +38,38 @@ def has_object_permission(self, request, view, obj):
             return super(IsPreprintFile, self).has_object_permission(request, view, obj.target)
 
         return True
+
+
+class RegistrationFileDetailPermission(permissions.BasePermission):
```

Comment:
So, I don't think this is actually necessary with the change to `get_file`. Sorry for the misleading ticket. I meant to update it after my explorations a couple weeks ago but didn't notice that this ticket actually got scheduled. >_<

Should be sufficient to just comment out the inclusion of this permission and re-run tests locally to see if that's right.